client.go

```diff
@@ -263,7 +265,6 @@ func (c *Client) run() {
 
 	defer func() {
 		c.conn.Close()
-		c.userCloseFunc()
 		close(c.userCloseWaitCh)
```

Comment:
This line needs to be moved too?

Also, can we have an unit test?

Reply:
Yes. I think UserOnCloseWait function may not necessary. Cause userCloseFunc() and rpc call are already synchronous in this situation. 
I think the problem mentioned in https://github.com/containerd/ttrpc/pull/68 can also be fixed by the current patch. rpc call will not return error until userCloseFunc() finish, then the Task.Delete goroutine works.

Reply:
If you have to wait for the closeFunc, you can call  `UserOnCloseWait`.  But the shim is killed unexpectedly and the `userCloseFunc()` can be called in async-defer. If we put the `userCloserFunc()` in `Close()`, no one will call it. Hope it can help.

Reply:
I didn't find a suitable place to add UserOnCloseWait. I find UserOnCloseWait is call in shim Delete, but this function will not be called when shim is killed unexpectedly.

Reply:
When the task is killed unexpectedly, the cleanup will be called and no need to sync the cleanup. The UserOnCloseWait is only be called when delete task manually because it needs sync.

Reply:
The UserOnCloseWait would not be invoked because ttrpc connect is closed。